File: data/readable_format/cpp/file2360.cpp
#include <iostream>
#include <bitset>

// Function to simulate the behavior of the Fortran IRBIT2 function
int IRBIT2(int &ISEED) {
    const int IB1 = 1;
    const int IB2 = 2;
    const int IB5 = 16;
    const int IB18 = 131072;
    const int MASK = IB1 + IB2 + IB5;

    if (ISEED & IB18) {
        ISEED = ((ISEED ^ MASK) << 1) | IB1;
        return 1;
    } else {
        ISEED = (ISEED << 1) & ~IB1;
        return 0;
    }
}

int main() {
    int seed = 12345; // Example seed value
    int result = IRBIT2(seed);
    
    std::cout << "Result: " << result << std::endl;
    std::cout << "New seed value: " << seed << std::endl;

    return 0;
}

File: data/readable_format/fortran/file2360.f90
PROGRAM test_irbit2
  IMPLICIT NONE
  INTEGER :: ISEED, result

  ! Example seed value
  ISEED = 12345

  ! Call the IRBIT2 function and print the result
  result = IRBIT2(ISEED)
  PRINT *, 'Result: ', result
  PRINT *, 'New seed value: ', ISEED

CONTAINS

  FUNCTION IRBIT2(ISEED)
    IMPLICIT NONE
    INTEGER :: IRBIT2
    INTEGER, INTENT(INOUT) :: ISEED
    INTEGER, PARAMETER :: IB1=1, IB2=2, IB5=16, IB18=131072
    INTEGER, PARAMETER :: MASK=IB1+IB2+IB5

    IF (IAND(ISEED, IB18) /= 0) THEN
      ISEED = IOR(ISHFT(IEOR(ISEED, MASK), 1), IB1)
      IRBIT2 = 1
    ELSE
      ISEED = IAND(ISHFT(ISEED, 1), NOT(IB1))
      IRBIT2 = 0
    ENDIF
  END FUNCTION IRBIT2

END PROGRAM test_irbit2